source/create.js

```diff
@@ -51,18 +53,19 @@ const create = defaults => {
 		return defaults.handler(url, options, next);
 	};
 
+	defaults.options.hooks = {
+		beforeRequest: [],
+		...(defaults.options.hooks || {})
+	};
+	got.hooks = defaults.options.hooks;
```

Comment:
Does this facilitate something? I'd avoid duplicating access points.

Reply:
`got.defaults.options.hooks` is duplicated. Consider why all options properties aren't copied to the got instance.

Reply:
No, it's not. Consider looking at the JS docs ;) Or google `object reference javascript`.

```js
let a = {};
let b = a;
a.c = 123;
console.log(a);
```

Reply:
You didn't parse my statement the way it was meant:

> I'd avoid duplicating access points.

I was saying you're creating more than one way of accessing the same object: `got.hooks` and `got.defaults.options.hooks`. The paths to access are duplicated.

Miscommunication and disagreement are bound to happen in async discourse, but we can figure things out without condescension. We can better understand each other by asking questions until the meaning is clear without presuming the other person is an idiot. 🤗

Reply:
> I'd avoid duplicating access points.

It's not named `access points`. There's no such thing in Node. I think you meant [`object references`](https://stackoverflow.com/questions/37290747/pass-by-reference-javascript-objects). Google it.

> I was saying you're creating more than one way of accessing the same object: got.hooks and got.defaults.options.hooks.

What's wrong with that? That's a shortcut. `got.defaults.options.hooks` is a long word, isn't it?

> We can better understand each other by asking questions until the meaning is clear without presuming the other person is an idiot.

Agreed. Using correct naming would end the argument. :tada: I've searched for `access points` but found nothing.

Reply:
Technical conversations are usually a mixture of colloquial and technical terms.

I think it's reasonable to expect to be able to converse like we're people, without being rude or disrespectful.

Apparently, those expectations are not going to be met here. :-(

Reply:
@szmarczak I know you don't mean anything bad by it, but your tone does come off a bit condescending. If you don't understand a comment, ask for clarification instead of assuming the other part doesn't know what they're talking about. It's easy to talk past each other. Clarity is key in technical conversations.

Reply:
@sindresorhus Yeah, you're right. My fault. I should better haven't said that. Sorry for being offensive.

@jstewmon Please take my sincere apologies.

> Technical conversations are usually a mixture of colloquial and technical terms.

On StackOverflow that behaviour is not welcome, just saying as I've experienced it myself.

Reply:
Thanks, no hard feelings